sys/kern/ringbuf.c

```diff
@@ -10,29 +10,32 @@ void ringbuf_init(ringbuf_t *rb, void *buf, size_t size) {
   rb->data = buf;
 }
 
-static void produce(ringbuf_t *buf, unsigned bytes) {
-  assert(buf->count + bytes <= buf->size);
-  assert(buf->head + bytes <= buf->size);
+/* below two function are to help to track ringbuf pointers in case when it is
+ * changed by something else. e.q. DMA */
+void ringbuf_produce(ringbuf_t *buf, unsigned bytes) {
+  assert(buf->count <= buf->size && bytes <= buf->size &&
+         buf->count <= buf->size - bytes);
```

Comment:
The first two statements are redundant, because they are just special cases of the third one. So:
`assert(buf->count <= buf->size - bytes);`
is sufficient.

Reply:
The first one should be already ok, but the second check is preventing us from underflow (unsigned).

Reply:
So we can simply reorder the last condition:`assert(bytes <= buf->size - buf->count);`, because we know that `buf->size >= buf->count`.

Reply:
Thanks, done